r_shadow.c

```diff
@@ -1226,7 +1226,7 @@ static unsigned int R_Shadow_MakeTextures_SamplePoint(float x, float y, float z)
 	float dist = sqrt(x*x+y*y+z*z);
 	float intensity = dist < 1 ? ((1.0f - dist) * r_shadow_lightattenuationlinearscale.value / (r_shadow_lightattenuationdividebias.value + dist*dist)) : 0;
 	// note this code could suffer byte order issues except that it is multiplying by an integer that reads the same both ways
-	return (unsigned char)bound(0, intensity * 256.0f, 255) * 0x01010101;
+	return ((unsigned int)bound(0, intensity * 256.0f, 255)) * 0x01010101U;
```

Comment:
Isn't this a pixel value that must be in the 0 to 255 range? So I'd be inclined to change the func return type to `unsigned char`. The type cast has higher precedence than the multiply.

Reply:
I'll do that.

Reply:
I'm quite confused on what this function is actually supposed to return... and what the purpose of the multiply is. I've moved the cast into the macro, should change its "return type" to `unsigned int` for multiplying with the integer.

Reply:
> I'm quite confused on what this function is actually supposed to return... and what the purpose of the multiply is.

Me too... looking at it again, it's not a pixel value.
0x01010101 is a bit bigger than the max number of colours representable in 24bit RGB?

Anyway if we keep the same behaviour and shadows work in Quake, then it should be fine.  

What tool are you using to catch this stuff?

Reply:
> Anyway if we keep the same behaviour and shadows work in Quake, then it should be fine.

Should be fine to keep it like this then.

> What tool are you using to catch this stuff?

[UBSan](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html).